### PR TITLE
add: quiz functionalities

### PR DESCRIPTION
**Pull request by thegeorgenikhil**
Added:
- Start quizzes
- Get questions from the API
- Send user answer to API returning the correct answer along with the score
- View results
- Private Routes

### CODE REVIEW

**Review by rahulbakshi4 — APPROVED** [review]
- The code structure is good
- The UI seems amazing

Suggestion: 
- Add a cursor pointer to the category cards and  to the options on question page
- The React quiz seems to be not working have a look at that 

Overall code looks fine to be merged, after considering fixes

**Review by mansipadgelwar — COMMENTED** [review]
unable to log in and signup

**Comment by thegeorgenikhil** [comment]
@rahulbakshi4 noted!

Yet to add questions for React that's why not working.

**Comment by thegeorgenikhil** [comment]
@mansipadgelwar try using this link https://deploy-preview-2--quizz-up-branch.netlify.app/

The other link doesn't seems to be working

**Review by mansipadgelwar — APPROVED** [review]

File: src/pages/QuizPage/QuizPage.css [hunk 1]
```diff
@@ -38,6 +38,19 @@
   padding: 1rem;
   margin: 10px 0;
```
Comment (samyakshah3008) [inline src/pages/QuizPage/QuizPage.css:39]: Good use of shorthand properties

File: src/pages/ResultPage/ResultPage.jsx [hunk 2]
```diff
@@ -1,248 +1,74 @@
...
+              <div className="quiz-info-container">
+                <p>
+                  Question: <span className="font-bold">{index + 1}</span>/
+                  <span className="font-bold">
+                    {results?.compareAnswer.length}
+                  </span>
+                </p>
+                <p>
+                  Score:{" "}
+                  <span className="font-bold">
```
Comment (samyakshah3008) [inline src/pages/ResultPage/ResultPage.jsx:43]: can remove this {" "}

**Review by samyakshah3008 — COMMENTED** [review]
Suggested few minor changes via single comments 
Rest is good for approval

**Pull request merged**